### ISSUE TITLE
Make configuration validation function return error message

### ISSUE THREAD

**Issue by spbnick** [spbnick/tlog#41]
Make tlog-rec and the future tlog-play configuration validation function return validation error string instead of printing it, so the function could be used in assertions.

**Comment by spbnick** [comment]
The first part of this was done with the introduction of error stacks, only the assertions need to be added now.

**Comment by spbnick** [comment]
This means asserting that `tlog_*_conf_validate` functions return TLOG_RC_OK on entering functions accepting corresponding configurations, and on exiting functions modifying them.

**Comment by danghai** [comment]
could you elaborate your idea? I see you check function `tlog_*_conf_validate` with return value `grc`. I am not sure about using `assertion` in here.

**Comment by spbnick** [comment]
Functions extracting data from loaded configuration expect it to be correct. Adding an `assert` to the start of the function verifying that, will help us ensure that. Similarly, functions which produce, or modify configuration, which are supposed to return a valid configuration need such `assert` at the exit.

**Comment by danghai** [comment]
I am little bit confused. the function `tlog_*_conf_validate` is used to check validation of configuration. It does not modify, or produce configuration. So I do not need to add `assert` in here. I put `assert` in the function that it is used. The commit shows an example about adding `assert` : 
https://github.com/danghai/tlog/commit/72205228a664b7d5006080f9c5d6e0bea2c62eac

**Comment by spbnick** [comment]
Yeah, we don't need to assert things in `tlog_*_conf_validate` functions for this issue, but instead use those functions to validate the configuration in functions which create, access or modify it. The `tlog_play_conf_file_load` doesn't need an assert, because it's the job of that function to validate the configuration and produce an error message for the user if it's wrong.

However, e.g. `tlog_play_conf_load` might need an `assert(tlog_play_conf_validate(NULL, conf, TLOG_CONF_ORIGIN_ARGS) == TLOG_RC_OK)` at the successful exit. I would add `TLOG_CONF_ORIGIN_MAX` (or `_LAST`, or `_TOP`), and use that instead, though.
